pkg/services/ngalert/schedule/schedule_unit_test.go

```diff
@@ -691,7 +691,7 @@ func setupScheduler(t *testing.T, rs *fakeRulesStore, is *state.FakeInstanceStor
 		Tracer:           testTracer,
 	}
 
-	st := state.NewManager(m.GetStateMetrics(), nil, is, &state.NoopImageService{}, mockedClock, &state.FakeHistorian{})
+	st := state.NewManager(m.GetStateMetrics(), nil, is, &state.NoopImageService{}, mockedClock, &state.FakeHistorian{}, false)
```

Comment:
Can we refactor the constructor args to a struct like `state.ManagerOptions`, kind of like we did for the scheduler? https://github.com/grafana/grafana/blob/2505f112f5e140e7ba45be82e27ecfc977a5aeea/pkg/services/ngalert/schedule/schedule.go#L104

The boolean argument that significantly changes the behavior is a bit confusing IMO.

Reply:
Oh, I actually like this approach. Will do it